eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/CompositeFastList.java

```diff
@@ -402,9 +401,8 @@ public int indexOf(Object o)
     {
         int offset = 0;
         int listsSize = this.lists.size();
-        for (int i = 0; i < listsSize; i++)
+        for (MutableList<E> list : this.lists)
```

Comment:
this is not the same: so-called-enhanced for loop creates garbage. (many more... any time there is a `list.get()`)